vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -991,4 +1091,77 @@ public Style getStyle() {
                 "Dialog does not support adding styles to overlay");
     }
 
+    static class OverlayClassListProxy extends AbstractSet<String>
+            implements ClassList {
+        private final HasStyle hasStyle;
+        private final ClassList classList;
+
+        public OverlayClassListProxy(HasStyle hasStyle) {
+            this.hasStyle = hasStyle;
+            this.classList = hasStyle.getElement().getClassList();
+        }
+
+        private void updateOverlayClass() {
+            hasStyle.getElement().setProperty("overlayClass",
+                    hasStyle.getClassName());
+        }
+
+        @Override
+        public Iterator<String> iterator() {
+            return classList.iterator();
```

Comment:
Removal through the iterator doesn't trigger `updateOverlayClass()` which might or might not be an acceptable compromise.

At the same time, implementing a custom iterator that handles removals in the "right" way would make it redundant to override most of the other methods in this class for any other purpose than to slightly optimize performance.